packages/deployments/contracts/contracts/messaging/connectors/axelar/BaseAxelar.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.17;
+
+import {AxelarExecutable} from "@axelar-network/axelar-gmp-sdk-solidity/contracts/executable/AxelarExecutable.sol";
+import {IAxelarGateway} from "@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IAxelarGateway.sol";
+import {IAxelarGasService} from "@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IAxelarGasService.sol";
+import {StringToAddress, AddressToString} from "@axelar-network/axelar-gmp-sdk-solidity/contracts/utils/AddressString.sol";
+
+abstract contract BaseAxelar is AxelarExecutable {
+  using StringToAddress for string;
+  using AddressToString for address;
+
+  // ============ Internal Storage ============
+  IAxelarGasService public immutable gasService;
+
+  // Mirror chain id
+  string public MIRROR_CHAIN_ID;
```

Comment:
this should be immutable

Reply:
string variable can't be immutable

Reply:
Should this be a string? I think we usually do unit32 for chain ID.

Reply:
> string variable can't be immutable

was afraid of that

> Should this be a string? I think we usually do unit32 for chain ID.

its not actually the chain id its the chain name